### PR TITLE
Add a CMake option to disable stub file generation, and fix stub generation on Python 3.6

### PR DESCRIPTION
**Pull request by philippewarren**

### CODE REVIEW

File: cmake/pip_utils.cmake [hunk 1]
```diff
@@ -74,32 +74,38 @@ endfunction()
 
 function(pip_add_stub_target)
     set(options)
-    set(oneValueArgs NAME PACKAGE_NAME)
-    set(multiValueArgs DEPENDS)
+    set(oneValueArgs NAME PACKAGE_NAME ENABLE_SO_STUB)
+    set(multiValueArgs DEPENDS STUB_DEPENDS)
```
Comment (philippewarren) [inline cmake/pip_utils.cmake:78]: DEPENDS will be passed to other targets if ENABLE_SO_STUB is false, while STUB_DEPENDS will not.
Same thing is done for HTML generation

File: cmake/pip_utils.cmake [hunk 2]
```diff
@@ -74,32 +74,38 @@ endfunction()
...
     cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
-    # Used via python -m to correctly add current directory to path, but can be checked for installation nonetheless
-    assert_program_installed("pybind11-stubgen")
-    string(REPLACE "." "/" PACKAGE_PATH ${ARGS_PACKAGE_NAME})
+    if(ARGS_ENABLE_SO_STUB)
+        # Used via python -m to correctly add current directory to path, but can be checked for installation nonetheless
+        assert_program_installed("pybind11-stubgen")
+        string(REPLACE "." "/" PACKAGE_PATH ${ARGS_PACKAGE_NAME})
```
Comment (philippewarren) [inline cmake/pip_utils.cmake:84]: Added guard, and moved check in guard

File: cmake/pip_utils.cmake [hunk 3]
```diff
@@ -74,32 +74,38 @@ endfunction()
...
-        VERBATIM
-    )
-    set(${ARGS_NAME} ${ARGS_NAME}-target ${PYTHON_PACKAGE_CONTENT_DIR}/__init__.pyi PARENT_SCOPE)
+        add_custom_command(
+            OUTPUT ${PYTHON_PACKAGE_CONTENT_DIR}/__init__.pyi
+            BYPRODUCTS ${PYTHON_PACKAGE_CONTENT_DIR}/py.typed
+            DEPENDS ${ARGS_DEPENDS} ${ARGS_STUB_DEPENDS}
+            # Required to allow generating a stub from the local version of the package, pybind11_stubgen changes the current directory
+            # before importing the module, as seen here: https://github.com/sizmailov/pybind11-stubgen/blob/master/pybind11_stubgen/__init__.py#L944-L945
+            COMMAND python3 -c "from pybind11_stubgen import main as gen; import os, sys; sys.path.insert(0, os.path.abspath('.')); gen(['${ARGS_PACKAGE_NAME}'])"
```
Comment (philippewarren) [inline cmake/pip_utils.cmake:92]: Fix weird Python 3.6 bug

File: cmake/pip_utils.cmake [hunk 4]
```diff
@@ -74,32 +74,38 @@ endfunction()
...
+        add_custom_target(
+            ${ARGS_NAME}-target
+            ALL
+            DEPENDS ${PYTHON_PACKAGE_CONTENT_DIR}/__init__.pyi
+            VERBATIM
+        )
+        set(${ARGS_NAME} ${ARGS_NAME}-target ${PYTHON_PACKAGE_CONTENT_DIR}/__init__.pyi PARENT_SCOPE)
+    else()
+        set(${ARGS_NAME} ${ARGS_DEPENDS} PARENT_SCOPE)
+    endif()
```
Comment (philippewarren) [inline cmake/pip_utils.cmake:108]: Pass dependencies if disabled

File: cmake/pip_utils.cmake [hunk 5]
```diff
@@ -108,9 +114,9 @@ function(pip_add_html_target)
     set(multiValueArgs DEPENDS DOC_DEPENDS)
     cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
-    assert_program_installed("sphinx-build")
-
     if(ARGS_ENABLE_HTML_DOC)
+        assert_program_installed("sphinx-build")
+
```
Comment (philippewarren) [inline cmake/pip_utils.cmake:119]: Moved check in guard

File: opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/package/setup.py.in [hunk 6]
```diff
@@ -50,8 +50,8 @@ setup(
     package_data={
         'opentera_webrtc.native_client': [
             "$<TARGET_FILE_NAME:${PYTHON_LIB_NAME}>",
-            "opentera_webrtc/native_client/py.typed",
-            "opentera_webrtc/native_client/__init__.pyi",
+            $<IF:$<BOOL:${ENABLE_SO_STUB}>,"opentera_webrtc/native_client/py.typed"$<COMMA>,>
+            $<IF:$<BOOL:${ENABLE_SO_STUB}>,"opentera_webrtc/native_client/__init__.pyi"$<COMMA>,>
```
Comment (philippewarren) [inline opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/package/setup.py.in:54]: Only add stub if it is generated

**Review by mamaheux — APPROVED** [review]

**Pull request merged**